core/src/main/scala/dagr/core/cmdline/parsing/ClpReflectiveBuilder.scala

```diff
@@ -130,24 +134,39 @@ private[parsing] class ClpArgument(declaringClass: Class[_],
   def omitFromCommandLine: Boolean = annotation.isEmpty
 
   if (annotation.isEmpty && defaultValue.isEmpty) {
-    throw new IllegalStateException(s"Must have a default value for arguments without annotations. Name: '$name' Declaring class: '${declaringClass.getSimpleName}'")
+    throw new IllegalStateException("Must have a default value for arguments without annotations. " +
+      s"Name: '$name' Declaring class: '${declaringClass.getSimpleName}'")
   }
 
   val mutuallyExclusive: mutable.Set[String] = new mutable.HashSet[String]()
-  annotation.foreach {_.mutex.foreach(mutuallyExclusive.add) }
-  val optional: Boolean = omitFromCommandLine || isFlag || hasValue || (isCollection && annotation.get.minElements() == 0) || argumentType == classOf[Option[_]]
+  annotation.foreach {
+    _.mutex.foreach(mutuallyExclusive.add)
+  }
+  val optional: Boolean = omitFromCommandLine ||
+    isFlag ||
+    hasValue ||
+    (isCollection && annotation.get.minElements() == 0) ||
+    argumentType == classOf[Option[_]]
 
   /** true if the field was set by the user */
   private[parsing] var isSetByUser: Boolean = false // NB: only true when [[setArgument]] is called, vs. this.value =
 
-  lazy val isSpecial: Boolean   = annotation.exists(_.special())
+  lazy val isSpecial: Boolean = annotation.exists(_.special())
   lazy val isSensitive: Boolean = annotation.exists(_.sensitive())
-  lazy val longName: String     = if (annotation.isDefined && annotation.get.name.nonEmpty) annotation.get.name else StringUtil.camelToGnu(name)
-  lazy val shortName: String    = annotation.map(_.flag()).getOrElse("")
-  lazy val doc: String          = annotation.map(_.doc()).getOrElse("")
-  lazy val isCommon: Boolean    = annotation.exists(_.common())
-  lazy val minElements: Int     = if (isCollection) annotation.map(_.minElements).getOrElse(1) else throw new IllegalStateException("Calling minElements on an argument that is not a collection.")
-  lazy val maxElements: Int     = if (isCollection) annotation.map(_.maxElements).getOrElse(Integer.MAX_VALUE) else throw new IllegalStateException("Calling minElements on an argument that is not a collection.")
+  lazy val longName: String = if (annotation.isDefined && annotation.get.name.nonEmpty) annotation.get.name else StringUtil.camelToGnu(name)
+  lazy val shortName: String = annotation.map(_.flag()).getOrElse("")
+  lazy val doc: String = annotation.map(_.doc()).getOrElse("")
+  lazy val isCommon: Boolean = annotation.exists(_.common())
+  lazy val minElements: Int = if (isCollection) {
```

Comment:
I don't love this layout but I'm struggling to come up with a better suggestion. Do we really need to throw an exception if this is called on a non-collection?  We could just return 0 or 1 depending on `required`...